resource-recorder/src/dal.rs

```diff
@@ -255,7 +256,11 @@ impl TryFrom<record_request::Resource> for Resource {
     type Error = String;
 
     fn try_from(value: record_request::Resource) -> Result<Self, Self::Error> {
-        Ok(Self::new(value.r#type.parse()?, value.data, value.config))
+        let r#type = value.r#type.parse()?;
+        if let Type::Custom = r#type {
+            return Err("Custom resources can not be saved in resource-recorder".into());
+        }
```

Comment:
There is nothing upstream (in deployer) that will trigger this error right?

Reply:
Deployers will send resources that trigger this, but they are filtered out here https://github.com/shuttle-hq/shuttle/pull/1543/files/ed61ad7423e45410df33936159cef20080204921#diff-7b2cbae17753959effc4bd23a9be282544990c7b84c18dc8f8e587282f0512a6R69-R71

Reply:
Great, thanks! :pray: